### DISCUSSION TITLE
Missing org/apache/logging/log4j/LogManager in the jar

### DISCUSSION THREAD

**Discussion by nvysyaraju-sift** [nielsbasjes/yauaa#896] [Q&A] [answered]
Hello,

While using either the fat jar with all transitive dependencies or normal jar for my testing and work, I am facing an issue with 
org/apache/logging/log4j/LogManager, which doesn't seem to be getting packaged into the jar, below is an example of the exception I am facing. What can I do here to have this class packaged in during the build?
```
java.lang.NoClassDefFoundError: org/apache/logging/log4j/LogManager
	at nl.basjes.parse.useragent.AnalyzerPreHeater.<clinit>(AnalyzerPreHeater.java:24)
```

**Comment by nielsbasjes** [comment, marked as answer]
Include the appropriate log4j dependencies.

**Reply by nvysyaraju-sift** [reply]
Yauaa uses log4j2 right, so should add appropriate dependency that matches with this.

**Reply by nielsbasjes** [reply]
Yes

**Comment by jsmith4467** [comment]
> Hello,
> 
> While using either the fat jar with all transitive dependencies or normal jar for my testing and work, I am facing an issue with org/apache/logging/log4j/LogManager, which doesn't seem to be getting packaged into the jar, below is an example of the exception I am facing. What can I do here to have this class packaged in during the build?
> 
> ```
> java.lang.NoClassDefFoundError: org/apache/logging/log4j/LogManager
> 	at nl.basjes.parse.useragent.AnalyzerPreHeater.<clinit>(AnalyzerPreHeater.java:24)
> ```

**Reply by jsmith4467** [reply]
Hi @nvysyaraju-sift,

I am getting similar log4j error. What are the dependencies you have included to resolve the issue.

Thanks

**Reply by nielsbasjes** [reply]
https://github.com/nielsbasjes/yauaa/blob/main/analyzer/src/it/Examples/java/pom.xml#L40

**Reply by nielsbasjes** [reply]
https://yauaa.basjes.nl/using/#logging-dependencies

**Reply by jsmith4467** [reply]
> https://yauaa.basjes.nl/using/#logging-dependencies

Thank you :)
